base/docs/helpdb/Dates.jl

```diff
@@ -49,11 +50,55 @@ All characters not listed above are treated as delimiters between date and time
 Dates.DateTime(dt::AbstractString, format::AbstractString)
 
 """
+    format(dt::TimeType, format::AbstractString; locale="english") -> AbstractString
+
+Construct a string by using a `TimeType` and applying the pattern provided by the
+`format` string. The following codes can be used for constructing the `format` string:
+
+| Code       | Example   | Comment                                                      |
+|:-----------|:----------|:-------------------------------------------------------------|
+| `y`        | 6         | Numeric year with a fixed width                              |
+| `m`        | 1, 12     | Numeric month with a minimum width                           |
+| `u`        | Jan       | Month name shortened to 3-chars according to the `locale`    |
+| `U`        | January   | Full month name according to the `locale` keyword            |
+| `d`        | 1, 31     | Day of the month with a minimum width                        |
+| `H`        | 0, 23     | Hour (24-hour clock) with a minimum width                    |
+| `M`        | 0, 59     | Minute with a minimum width                                  |
+| `S`        | 0, 59     | Second with a minimum width                                  |
+| `s`        | 000, 500  | Millisecond with a minimum width of 3                        |
+| `e`        | Mon, Tue  | Abbreviated days of the week                                 |
+| `E`        | Monday    | Full day of week name                                        |
+
+The number of sequential code characters indictate the width of the code. A format of
+`yyyy-mm` specifies that the code `y` should have a width of four while `m` a width of two.
+Codes yielding a numeric string will have associated width mode: fixed or minimum. Fixed
```

Comment:
I definitely was having the most difficulty explaining this section. Basically only the codes that produce a numeric value (`y`, `m`, ...) care about the width. Currently the width can be treated in two different ways. Here's a good example that I need to figure out how to incorporate:
```julia
julia> Dates.format(Date(1996,1,2), "yyyy-mm")
"1996-01"

julia> Dates.format(Date(1996,12,2), "yyyy-mm")
"1996-12"

julia> Dates.format(Date(1996,1,2), "yy-m")
"96-1"

julia> Dates.format(Date(1996,12,2), "yy-m")
"96-12"
```